app/src/main/kotlin/model/graphs/abstractGraph/Graph.kt

```diff
@@ -0,0 +1,93 @@
+package model.graphs.abstractGraph
+
+import kotlin.NoSuchElementException
+import kotlin.collections.ArrayList
+
+abstract class Graph<D> {
+    protected val vertices: ArrayList<Vertex<D>> = arrayListOf()
```

Comment:
We wanted to make a "get a vertex knowing its index" operation to be as quick as possible, so algorithms can initially work with indices and take vertices from array very fast. But it seems that our efforts are of no use because `getVertices()` returns a list...